nexosim/src/server/run.rs

```diff
@@ -28,7 +28,7 @@ use super::services::{
 /// public event and query interface.
 pub fn run<F, I>(sim_gen: F, addr: SocketAddr) -> Result<(), Box<dyn std::error::Error>>
 where
-    F: FnMut(I) -> InitResult + Send + 'static,
+    F: FnMut(I) -> SimInit + Send + 'static,
```

Comment:
As suggested earlier, I would be for making this fallible, but since this would probably need to be reflected in the gRPC API, let;'s move this to a separate PR.